app/clients/hedera/client.go

```diff
@@ -125,50 +124,69 @@ func (hc Node) SubmitScheduledTokenBurnTransaction(tokenID hedera.TokenID, amoun
 // SubmitTopicConsensusMessage submits the provided message bytes to the
 // specified HCS `topicId`
 func (hc Node) SubmitTopicConsensusMessage(topicId hedera.TopicID, message []byte) (*hedera.TransactionID, error) {
-	tx, err := hedera.NewTopicMessageSubmitTransaction().
-		SetTopicID(topicId).
-		SetMessage(message).
-		SetMaxRetry(hc.maxRetry).
-		FreezeWith(hc.client)
+	attempt := 0
+	for attempt <= hc.maxRetry {
+		attempt++
+		tx, err := hedera.NewTopicMessageSubmitTransaction().
+			SetTopicID(topicId).
+			SetMessage(message).
+			SetMaxRetry(hc.maxRetry).
+			FreezeWith(hc.client)
 
-	if err != nil {
-		return nil, err
-	}
+		if err != nil {
+			return nil, err
+		}
 
-	hc.logger.Debugf("Submit Topic Consensus Message, with transaction ID [%s] and Node Account IDs: %v",
-		tx.GetTransactionID(),
-		tx.GetNodeAccountIDs(),
-	)
+		hc.logger.Debugf("Submit Topic Consensus Message, with transaction ID [%s] and Node Account IDs: %v",
+			tx.GetTransactionID(),
+			tx.GetNodeAccountIDs(),
+		)
 
-	response, err :=  tx.Execute(hc.GetClient())
+		response, err := tx.Execute(hc.GetClient())
+		if shouldRetryTransaction(err) {
+			continue
+		}
 
-	if err != nil {
-		return nil, err
-	}
+		if err != nil {
+			return nil, err
+		}
 
-	_, err = hc.checkTransactionReceipt(response)
+		_, err = hc.checkTransactionReceipt(response)
 
-	return &response.TransactionID, err
+		return &response.TransactionID, err
+	}
+
+	return nil, fmt.Errorf("Too many retries to submit topic message to %s", topicId.String())
```

Comment:
I believe this should be logged with `hc.logger.Debugf(...)`

Reply:
I definitely think this should be an error as we have an unfinished transaction.